subsys/net/lib/nrf_cloud/src/nrf_cloud_codec_internal.c

```diff
@@ -2528,11 +2531,12 @@ int nrf_cloud_gnss_msg_json_encode(const struct nrf_cloud_gnss_data * const gnss
 	/* On failure, remove any items added to the provided object */
 	cJSON_DeleteItemFromObject(gnss_msg_obj, NRF_CLOUD_JSON_APPID_KEY);
 	cJSON_DeleteItemFromObject(gnss_msg_obj, NRF_CLOUD_JSON_MSG_TYPE_KEY);
-	cJSON_DeleteItemFromObject(gnss_msg_obj, NRF_CLOUD_JSON_DATA_KEY);
 	cJSON_DeleteItemFromObject(gnss_msg_obj, NRF_CLOUD_MSG_TIMESTAMP_KEY);
-	/* Cleanup data_obj if it wasn't added to gnss_msg_obj */
+
 	if (data_obj) {
 		cJSON_Delete(data_obj);
+	} else {
+		cJSON_DeleteItemFromObject(gnss_msg_obj, NRF_CLOUD_JSON_DATA_KEY);
 	}
```

Comment:
that will not work for non NRF_CLOUD_GNSS_TYPE_PVT types.

Reply:
Oh! You're right -- but it could work if you do:

```c
/* Add the NMEA sentence to the message */
data_obj = cJSON_AddStringToObject(gnss_msg_obj, NRF_CLOUD_JSON_DATA_KEY, nmea);

if (!data_obj) {
	ret = -ENOMEM;
	goto cleanup;
}
```

Which might be nice since it would make the memory management consistent between both type groups

Reply:
fixed in a different way.

Reply:
I dig it, that's even cleaner